### DISCUSSION TITLE
Feature request: add jitter to refresh interval of individual panels

### DISCUSSION THREAD

**Discussion by JorritSalverda** [grafana/grafana#36991] [Dashboards, Panels & Visualizations]
To distribute load more evenly on a queried system like ElasticSearch or Prometheus it will help to add a little bit of **randomness / jitter** to the **refresh interval** of **each individual panel** and every time it triggers pick a new 'random' interval until the next update. 

With the current implementation of very rigid intervals our Prometheus server has huge spikes in cpu load corresponding with the refresh intervals of frequently used dashboards.

See article https://cloudplatform.googleblog.com/2016/11/how-to-avoid-a-self-inflicted-DDoS-Attack-CRE-life-lessons.html about why this is useful.

An implementation in Javascript might look like this for a jitter percentage of 25% resulting in a random number between 75% and 125% of the defined interval for the respective dashboard:

```
jitteredIntervalUntilNextRefresh = Math.random() * 0.5 * interval + 0.75 * interval  
```

Most important is that every panel calculates it's own interval until the next refresh every time, to prevent it from slowly synchronizing after time and does so after each refresh.

**Comment by mtanda** [comment]
To support this, we may need to consider align time range.
https://github.com/grafana/grafana/issues/8235

Without it, graph time range is not same among all graphs.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83019. Feel free to continue the discussion around this there. Thank you!
